### PR TITLE
Re dynamorevit linkelementnodes

### PR DESCRIPTION
**Pull request by dariaivanciucova**
### Purpose

Re-do of [DynamoRevit LinkElementNodes](https://github.com/DynamoDS/DynamoRevit/pull/2910) PR. 

Note! Currently targeting Revit 2024 dlls, to be able to build. There are changes inside multiple project files where the references have been updated. Potentially these need to be removed from the PR before merging, but on a few occasions, there are valid changes to the project files that need to be kept.

This PR contains new nodes for Dynamo Revit - all falling under LinkElement category.
image
image

- LinkElement.GetGeometry
- LinkElement.GetLocation
- LinkElement.BoundingBox
- LinkElement.LinkInverseTransform
- LinkElement.LinkTransform
- LinkElement.ByRayBounce

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@Amoursol
@hwahlstrom

### FYIs

@dnenov

### CODE REVIEW

**Comment by dariaivanciucova** [comment]
Note! Currently targeting Revit 2024 dlls, to be able to build. There are changes inside multiple project files where the references have been updated. Potentially these need to be removed from the PR before merging, but on a few occasions, there are valid changes to the project files that need to be kept.

**Comment by dariaivanciucova** [comment]
> 

This should now be resolved

File: src/DynamoRevit/DynamoRevit.csproj
Comment (reddyashish) [inline src/DynamoRevit/DynamoRevit.csproj:57]: This one needs to be removed still

**Review by reddyashish — APPROVED** [review]

**Comment by dariaivanciucova** [comment]
@reddyashish tested - all works as expected

File: src/Libraries/RevitNodes/Elements/LinkElement.cs
Comment (Mikhinja) [inline src/Libraries/RevitNodes/Elements/LinkElement.cs:35]: What if, by accident, the `linkElement` given is not actually a link element, but an element in the current Revit document?

File: src/Libraries/RevitNodes/Elements/LinkElement.cs
Comment (Mikhinja) [inline src/Libraries/RevitNodes/Elements/LinkElement.cs:67]: What if, by accident, the `linkElement` given is not actually a link element, but an element in the current Revit document?

Comment (dariaivanciucova) [reply src/Libraries/RevitNodes/Elements/LinkElement.cs:67]: @Mikhinja the ZoomToLinkedElement only gets applied where the element's document does not equal the current document. 
```
                Document elementDoc = element.InternalElement.Document;
                Document currrentDoc = Revit.Application.Document.Current.InternalDocument;
                if (!(elementDoc.Equals(currrentDoc)) && (element.InternalElement.IsValidObject))
                {
                    Revit.Elements.LinkElement.ZoomToLinkedElement(element);
```
that otta do it, right? do you think there are any cases where this wouldn't work?

Comment (Mikhinja) [reply src/Libraries/RevitNodes/Elements/LinkElement.cs:67]: @dariaivanciucova yes, that otta do it

Comment (Mikhinja) [reply src/Libraries/RevitNodes/Elements/LinkElement.cs:67]: @dariaivanciucova Sorry for the pause in replies.
You don't seem to need the `location` here, should that line perhaps be removed?

**Comment by reddyashish** [comment]
@Mikhinja Can we merge this if it looks good.

File: src/Libraries/RevitNodes/Elements/LinkElement.cs
Comment (dariaivanciucova) [reply src/Libraries/RevitNodes/Elements/LinkElement.cs]: hey @Mikhinja done!

**Review by Mikhinja — APPROVED** [review]

**Pull request merged**
